cpp/src/arrow/util/macros.h

```diff
@@ -19,6 +19,8 @@
 
 #include <cstdint>
 
+#include "arrow/util/simd.h"
```

Comment:
I don't think we should be including this here? `macros.h` is a very light-weight header that should probably not include other Arrow headers.

Reply:
OK. This is because we want to have optimal prefetch (the _mm_prefetch intrinsic) for MS systems on simd-equipped CPUs. 

Shall I move the whole prefetch definitions to another individual header? So that macros.h remains lightweight.

Reply:
Ah, well, that makes sense.